src/Java.Interop.Tools.Maven/Repositories/MavenRepository.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Net.Http;
+using Java.Interop.Tools.Maven.Models;
+
+namespace Java.Interop.Tools.Maven.Repositories;
+
+public class MavenRepository : IMavenRepository
+{
+	readonly string base_url;
+	readonly HttpClient client;
+
+	public string Name { get; }
+
+	public MavenRepository (string baseUrl, string name)
+	{
+		Name = name;
+		base_url = baseUrl;
+
+		client = new HttpClient {
```

Comment:
*General* [guidelines for `HttpClient` usage](https://learn.microsoft.com/en-us/dotnet/fundamentals/networking/http/httpclient-guidelines#recommended-use) is that `HttpClient` instances should either be long-lived and shared (and `static`) *or* through [`IHttpClientFactory`](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.ihttpclientfactory) (though I'm not quite sure what *that* actually looks like in practice…).

`new`ing up and ~immediately disposing of `HttpClient` instances is not a good idea.

See also: https://learn.microsoft.com/en-us/dotnet/fundamentals/runtime-libraries/system-net-http-httpclient#instancing

Reply:
Related: [`HttpClient.BaseAddress`](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.baseaddress?view=net-8.0#system-net-http-httpclient-baseaddress) is a read+write property.

Reply:
This shouldn't have been too bad, as `MavenRepository` itself is generally expected to be long-lived, but switched to using a `static` `HttpClient` to be better.